dialog_flow_chatgpt_skill/dff_chatgpt_skill/processing.py

```diff
@@ -0,0 +1,158 @@
+        return ctx
+
+    return extract_intents_inner
+
+
+def clear_intents():
+    """
+    Clear intents container.
+    """
+
+    def clear_intents_inner(ctx: Context, _: Actor) -> Context:
+        ctx.misc[utils.DNNC_INTENTS] = []
+        return ctx
+
+    return clear_intents_inner
+
+
+def clear_slots():
+    """
+    Clear slots container.
+    """
+
+    def clear_slots_inner(ctx: Context, _: Actor) -> Context:
+        ctx.misc[utils.SLOTS] = {}
+        return ctx
+
+    return clear_slots_inner
+
+
+def generate_response():
+    """
+    Store ChatGPT output and ChatGPT coherence measure in the context.
+    """
+    expression = re.compile(r"true", re.IGNORECASE)
+
+    def generate_response_inner(ctx: Context, _: Actor) -> Context:
+        if ctx.validation:
+            return ctx
+
+        bot = Chatbot(config=CONFIG)
+        text = ctx.last_request.text
+        full_prompt = " ".join([utils.CHATGPT_MAIN_PROMPT, text])
+        chatgpt_output = send_message(bot, full_prompt)
+        full_coherence_prompt = " ".join([utils.CHATGPT_COHERENCE_PROMPT, text, chatgpt_output])
+        coherence_output = send_message(bot, full_coherence_prompt)
+        ctx.misc[utils.CHATGPT_OUTPUT] = chatgpt_output
+        ctx.misc[utils.CHATGPT_COHERENCE] = True if re.search(expression, coherence_output) else False
+        return ctx
+
+    return generate_response_inner
+
+
+def extract_item():
+    """
+    Extract item slot.
+    """
+    expression = re.compile(r".+")
+
+    def extract_item_inner(ctx: Context, _: Actor) -> Context:
+        if ctx.validation:
+            return ctx
+
+        text: str = ctx.last_request.text
+        search = re.search(expression, text)
+        if search is not None:
+            group = search.group()
+            ctx.misc[utils.SLOTS]["items"] = group.split(", ")
+        return ctx
+
+    return extract_item_inner
+
+
+def extract_payment_method():
+    """Extract payment method slot."""
+    expression = re.compile(r"(card|cash)", re.IGNORECASE)
+
+    def extract_payment_method_inner(ctx: Context, _: Actor) -> Context:
+        if ctx.validation:
+            return ctx
+
+        text: str = ctx.last_request.text
+        search = re.search(expression, text)
+        if search is not None:
+            ctx.misc[utils.SLOTS]["payment_method"] = search.group()
+        return ctx
+
+    return extract_payment_method_inner
+
+
+def extract_delivery():
+    """
+    Extract delivery slot.
+    """
+    expression = re.compile(r"(pickup|deliver)", re.IGNORECASE)
+
+    def extract_delivery_inner(ctx: Context, _: Actor) -> Context:
+        if ctx.validation:
+            return ctx
+
+        text: str = ctx.last_request.text
+        search = re.search(expression, text)
+        if search is not None:
+            ctx.misc[utils.SLOTS]["delivery"] = search.group()
+        return ctx
+
+    return extract_delivery_inner
+
+
+def extract_address():
```

Comment:
Either remove this method or use it in the script:

- inside a node that asks for delivery address only if user picked "delivery"
- inside a node that asks for address and in case the user picked "pickup" sends the closest pickup location (inside the confirmation message)

Could also do this: if the user picks delivery, bot prompts for an address; if user picks pickup, bot gives a list of pickup locations as buttons and expects a user to click one of them.

Reply:
Done